test/e2e/validation_test.go

```diff
@@ -289,13 +288,31 @@ func (tc *testContext) sshSetup() error {
 	return nil
 }
 
-// runSSHJob creates and waits for a Kubernetes job to run. The command provided will be executed on the host specified
-// by the provided IP.
-func (tc *testContext) runSSHJob(name, command, ip string) (string, error) {
+// runPowershellSSHJob creates and waits for a Kubernetes job to run. The command provided will be executed through
+// powershell, on the host specified by the provided IP.
+func (tc *testContext) runPowershellSSHJob(name, command, ip string) (string, error) {
+	keyMountDir := "/private-key"
+	sshCommand := []string{"bash", "-c",
+		fmt.Sprintf(
+			// first determine if the host has powershell or cmd as the default shell
+			"PSVERSION_TABLE=$(ssh -o StrictHostKeyChecking=no -i %s %s@%s 'echo ($PSVersionTable)');"+
```

Comment:
This looks really fragile. There must be a cleaner way to figure out what the default shell is.

Reply:
Since we cannot check the default shell in the operator code through the registry, unless we want to force users to use OpenSSH, I'd prefer to keep the test code consistent with what we are doing there.